web/src/routes/(user)/share/[key]/+page.svelte

```diff
@@ -1,20 +1,77 @@
 <script lang="ts">
   import AlbumViewer from '$lib/components/album-page/album-viewer.svelte';
   import IndividualSharedViewer from '$lib/components/share-page/individual-shared-viewer.svelte';
-  import { SharedLinkType } from '@api';
+  import ControlAppBar from '$lib/components/shared-components/control-app-bar.svelte';
+  import ImmichLogo from '$lib/components/shared-components/immich-logo.svelte';
+  import ThemeButton from '$lib/components/shared-components/theme-button.svelte';
+  import Button from '$lib/components/elements/buttons/button.svelte';
+  import { api, SharedLinkType } from '@api';
   import type { PageData } from './$types';
+  import { handleError } from '$lib/utils/handle-error';
 
   export let data: PageData;
-  const { sharedLink } = data;
+  let { sharedLink, passwordRequired, sharedLinkKey: key } = data;
 
-  let isOwned = data.user ? data.user.id === sharedLink.userId : false;
+  let isOwned = false;
+  let password = '';
+
+  const handlePasswordSubmit = async () => {
+    try {
+      const result = await api.sharedLinkApi.getMySharedLink({ password, key });
+      passwordRequired = false;
+      sharedLink = result.data;
+      isOwned = data.user ? data.user.id === sharedLink.userId : false;
+      document.title = (sharedLink.album ? sharedLink.album.albumName : 'Public Share') + ' - Immich';
+      document
+        .querySelector('meta[name="description"]')
+        ?.setAttribute('content', sharedLink.description || `${sharedLink.assets.length} shared photos & videos.`);
+      document.cookie = `immich_shared_link_token=${sharedLink.token}; path=/; max-age=86400; samesite=Lax`;
```

Comment:
This could be an http only cookie set by the server instead.

Reply:
Should it be set on the server？

Reply:
Yes.